cirq-google/cirq_google/engine/engine_processor.py

```diff
@@ -119,12 +330,17 @@ def get_device(self, gate_sets: Iterable[SerializableGateSet]) -> cirq.Device:
         spec = self.get_device_specification()
         if not spec:
             raise ValueError('Processor does not have a device specification')
-        return serializable_device.SerializableDevice.from_proto(spec, gate_sets)
+        if not all(isinstance(gs, SerializableGateSet) for gs in gate_sets):
+            raise ValueError('All gate_sets must be SerializableGateSet currently.')
```

Comment:
Normally we don't want to make type checking that's outside the API type boundaries of the function part of the function itself (thereby growing the function to now have it's API include things outside the intended type boundaries). Do we have a good reason here for adding this check ?